great_expectations_cloud/agent/run.py

```diff
@@ -1,23 +1,47 @@
 from __future__ import annotations
 
+import asyncio
 import logging
-from typing import Final
+from typing import Any, Final
 
-from great_expectations_cloud.agent import GXAgent
-from great_expectations_cloud.agent.agent import GXAgentConfigError
+from fast_depends import (
+    Depends,  # noqa: TCH002 - WARNING: This is used for a type hint, but pydantic will fail if not imported this way
+)
+from faststream import Context, FastStream
+from faststream.rabbit import RabbitBroker
+from typing_extensions import (
+    Annotated,  # noqa: TCH002 - WARNING: This is used for a type hint, but pydantic will fail if not imported this way
+)
+
+from great_expectations_cloud.agent.agent import GXAgent, agent_instance
+from great_expectations_cloud.agent.config import MAX_DELIVERY, GXAgentConfig, GXAgentConfigError
+from great_expectations_cloud.agent.queue import declare_queue
 
 LOGGER: Final[logging.Logger] = logging.getLogger(__name__)
 
+# Note: faststream requires module level variables for its testing implementation
+config = GXAgentConfig.build()
+broker = RabbitBroker(url=str(config.connection_string))
+app = FastStream(broker)
+queue = declare_queue(config)
+
+
+@broker.subscriber(queue, retry=MAX_DELIVERY)  # type: ignore[arg-type]
```

Comment:
Hi there! You made a great job! But, I have to warn you:
We are planning to remove `retry` option in **FastStream 0.6.0**, so you should refactor this place to use `message.ack/nack/reject` logic explicitly